### PR TITLE
Create Application Administrator

### PR DESCRIPTION
**Pull request by ross-p-smith**
# PR for issue 1716

## What is being addressed

Amend the `make auth` process to create a Service Principal with Application.ReadWrite.All

## How is this addressed

- Create new bash script to create the service principal.
- Chosen not to make the existing script anymore complicated than it needs to be
- Removed any Application.ReadWrite.All logic from the API App

### CODE REVIEW

**Comment by ross-p-smith** [comment]
/test-force-approve

There are no tests for this area. You would need to run `make auth` with AUTO_WORKSPACE_APP_REGISTRATION set to true in your env file

File: scripts/aad/create_application_administrator.sh [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+fi
+
+# Get the directory that this script is in
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+declare grantAdminConsent=0
+declare currentUserId=""
+declare spId=""
+declare msGraphUri="https://graph.microsoft.com/v1.0"
+declare appName="sp_application_admin"
```
Comment (marrobi) [inline scripts/aad/create_application_administrator.sh:38]: Feel this should be TRE specific so know which TRE carried out the operation.

**Review by marrobi — APPROVED** [review]
LGTM - one comment re naming.

File: scripts/aad/create_application_administrator.sh (hunk 1 above)
Comment (ross-p-smith) [reply scripts/aad/create_application_administrator.sh]: Done!

**Pull request merged**